### DISCUSSION TITLE
RoutiSigHandler

### DISCUSSION THREAD

**Discussion by hemalbavishi** [eclipse-iceoryx/iceoryx#1515] [Q&A] [answered]
During my analysis, I found bit strange piece of code in roudi_app.cpp

![image](https://user-images.githubusercontent.com/41425844/179232591-4805df91-a5f8-40b3-a3e4-f02c867adc39.png)

I have a question on this piece of code
What's the purpose of g_RoudiApp in registerSigHandler function and roudiSigHandler function? Do we really need this global variable?

**Comment by elfenpiff** [comment, marked as answer]
@hemalbavishi You are right, this semaphore is now superfluous that is why we removed it on master some time ago. The new way is to use the posix signalwatcher from iceoryx hoofs when one would like to wait until the (roudi) app received a `SIGTERM` signal.
See:
```cxx
#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"

//... some code

iox::posix::waitForTerminationRequest();
```

**Reply by hemalbavishi** [reply]
@elfenpiff : Thank you for your quick response.  Looks like I am using bit older code, hence I raised this issue. I will sync up with master code.
